hummingbot/connector/exchange/kucoin/kucoin_exchange.py

```diff
@@ -351,7 +351,65 @@ async def _update_trading_fees(self):
             trading_pair = await self.trading_pair_associated_to_exchange_symbol(symbol=fee_json["symbol"])
             self._trading_fees[trading_pair] = fee_json
 
+    async def _update_orders_fills(self, orders: List[InFlightOrder]):
+        # This method in the base ExchangePyBase, makes an API call for each order.
+        # Given the rate limit of the API method and the breadth of info provided by the method
+        # the mitigation proposal is to collect all orders in one shot, then parse them
+        # Note that this is limited to 50000 orders
+        # An alternative for Kucoin would be to use the limit/fills that returns 24hr updates, which should
+        # be sufficient, the rate limit seems better suited
+        all_trades_updates: List[TradeUpdate] = []
+        try:
+            all_trades_updates: List[TradeUpdate] = await self._all_trades_updates()
+        except asyncio.CancelledError:
+            raise
+        except Exception as request_error:
+            self.logger().warning(
+                f"Failed to fetch trade updates. Error: {request_error}")
+
+        for order in orders:
+            # Matching the orders based on their exchange_order_id
+            trade_updates = [t for t in all_trades_updates if t.exchange_order_id == order.exchange_order_id]
+            for trade_update in trade_updates:
+                # We need to update the client_order_id and trading pair
+                trade_update.client_order_id = order.client_order_id
+                trade_update.trading_pair = order.trading_pair
+                self._order_tracker.process_trade_update(trade_update)
+
+    async def _all_trades_updates(self) -> List[TradeUpdate]:
+        trade_updates: List[TradeUpdate] = []
+        all_fills_response = await self._api_get(
+            path_url=CONSTANTS.ORDER_FILLS_URL,
+            params={
```

Comment:
create a new variable in the class that will be: last_order_fills_request_ts.
initialize it in 0 and here create a new variable:
next_start_time = max(self.last_order_fills_request_ts, lower creation timestamp of the orders that you are trying to use)
after the query assign the last timestamp received of the trades to the last_order_fills_request_ts

Reply:
For this, you're saying to only take the trades that have a 'start' more recent than the 'next_start_time', correct?

Reply:
Got it